vaadin-renderer-flow-parent/vaadin-renderer-flow/src/test/java/com/vaadin/flow/data/renderer/FormattedRenderersSerializableTest.java

```diff
@@ -0,0 +1,34 @@
+package com.vaadin.flow.data.renderer;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.ObjectOutputStream;
+
+import org.junit.Test;
+
+public class FormattedRenderersSerializableTest {
+
+    @Test
+    public void numberRendererIsSerializable() throws IOException {
+        final NumberRenderer renderer = new NumberRenderer(
+                value -> value.toString(), "");
+        new ObjectOutputStream(new ByteArrayOutputStream())
+                .writeObject(renderer);
+    }
+
+    @Test
+    public void localDateTimeRendererIsSerializable() throws IOException {
+        final LocalDateTimeRenderer renderer = new LocalDateTimeRenderer(
+                value -> value.toString(), "");
+        new ObjectOutputStream(new ByteArrayOutputStream())
+                .writeObject(renderer);
+    }
+
+    @Test
+    public void localDateRendererIsSerializable() throws IOException {
+        final LocalDateRenderer renderer = new LocalDateRenderer(
+                value -> value.toString(), "");
+        new ObjectOutputStream(new ByteArrayOutputStream())
+                .writeObject(renderer);
+    }
+}
```

Comment:
Differently from the test for `NumberRenderer`, both tests turn red if I rollback the changes on the respective renderers. However, trying with the example code shared on the original issue, the tests become red with `NotSerializableException`.

The difference between the code is that they use different constructors:
 - Current test uses the constructor that accepts a string as the format pattern which is being used to generate the formatter inside the supplier callback (eg. `LocalDateRenderer(ValueProvider<SOURCE, LocalTime> valueProvider, String formatPattern)`)
 - The snippet provided by the issue reporter uses the constructor that receives the `DateTimeFormatter` object directly, which is then used by the supplier callback (eg. `LocalDateRenderer(ValueProvider<SOURCE, LocalDate> valueProvider, DateTimeFormatter formatter)`)
 
The main difference between them is that as the object reference lives outside the supplier, the object can't be serialized.

Reply:
Good catch. This means that introducing new constructors and deprecating the old ones is actually mandatory to fix this. I just tried the change locally with improved unit test and that way it works.